proposals/dao/fip_30b.ts

```diff
@@ -0,0 +1,51 @@
+import hre, { ethers } from 'hardhat';
+import { expect } from 'chai';
+
+async function setup(addresses, oldContracts, contracts, logging) {}
+
+/*
+ 1. Grant Governor to FeiDAOTimelock 
+ 2. Grant PCVController to FeiDAOTimelock
+ 3. Revoke PCVController from old Timelock
+ 4. Grant Minter to FeiDAOTimelock
+ 5. Revoke Minter from old Timelock
+ 6. Transfer 1M kashi DPI LP tokens to OA Timelock
+ 7. Transfer 2.5M kashi ETH LP tokens to OA Timelock
+ 8. Transfer 2.5M kashi TRIBE LP tokens to OA Timelock
+ 9. Transfer 2.5M kashi xSUSHI LP tokens to OA Timelock
+*/
+async function run(addresses, oldContracts, contracts, logging = false) {}
+
+async function teardown(addresses, oldContracts, contracts, logging) {}
+
+async function validate(addresses, oldContracts, contracts) {
```

Comment:
type these functions appropriately, eg ValidateUpgradeFunc

Reply:
done

Reply:
you forgot to type validate